src/chunkserver/raftlog/curve_segment.cpp

```diff
@@ -604,12 +603,16 @@ int CurveSegment::close(bool will_sync) {
     if (_last_index > _first_index) {
         if (FLAGS_raftSyncSegments && will_sync &&
                                 !FLAGS_enableWalDirectWrite) {
-            ret = braft::raft_fsync(_fd);
+            if (braft::FLAGS_raft_use_fsync_rather_than_fdatasync) {
```

Comment:
This means that fdatasync can be used when using filepool, right? Why add another flag? Don't use the walpool related configuration in the configuration file directly.

Reply:
This logic is same as  braft::raft_fsync